presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java

```diff
@@ -158,50 +169,45 @@ public boolean advanceNextPosition()
         {
             while (true) {
                 if (cursorOffset >= split.getEnd()) {
-                    return endOfData(); // Split end is exclusive.
+                    return endOfData(1); // Split end is exclusive.
```

Comment:
It means that the offset has reached or gone beyond the end of this split. A split in Kafka sense is just a range specified by 2 message offsets.